kliff/dataset/dataset.py

```diff
@@ -138,6 +169,122 @@ def to_file(self, filename: Path, file_format: str = "xyz"):
                 f"got: {file_format}."
             )
 
+    @classmethod
+    def from_colabfit_dataobjects(
+        cls,
+        database_client: "MongoDatabase",
+        configuration_id: str,
+        property_ids: Optional[Union[List[str], str]] = None,
+        weight: Optional[Weight] = None,
+    ):
+        """
+        Read configuration from colabfit database .
+
+        Args:
+            database_client: Instance of connected MongoDatabase client, which can be used to
+            fetch database from colabfit-tools dataset.
+            configuration_id: ID of the configuration instance to be collected from the collection
+            "configuration" in colabfit-tools.
+            property_ids: ID of the property instance to be associated with current configuration.
+            Usually properties would be trained against. Each associated property "field" will be
+            matched against provided list of aux_property_fields.
+             weight:
+        """
+        try:
+            fetched_configuration: "ColabfitConfiguration" = (
```

Comment:
No. `colabfit.tools` also have a class called `Configuration`, so to disambiguate, I imported it as `ColabfitConfiguration`

Reply:
OK, then let's annotate it with the real class  `ColabfitConfiguration` but not the string.

Reply:
It was annotated using typing [annotation](https://docs.python.org/3.9/library/typing.html) strings and `TYPE_CHECKING` condition, as otherwise colabfit will become a runtime dependency and raise error if colabfit-tools is not installed. I am not sure that many users will use colabfit to justify making it a hard dependency. Therefore "ColabfitConfiguration" is a string. If you want to keep style consistent etc, Then I would recommend either removing type annotations to ColabfitConfiguration completely (preferable) or creating a dummy object.

Reply:
OK, then using a string annotation makes sense.

Reply:
OK. Keeping the string based annotations for now.